pkg/daemon/ceph/osd/volume.go

```diff
@@ -641,6 +641,29 @@ func (a *OsdAgent) initializeDevicesLVMMode(context *clusterd.Context, devices *
 				if device.Config.MetadataDevice != "" {
 					md = device.Config.MetadataDevice
 				}
+				var metadataDevice *sys.LocalDisk
+			OUTER:
+				for _, localDevice := range context.Devices {
+					if localDevice.Name == md || filepath.Join("/dev", localDevice.Name) == md {
+						logger.Infof("%s found in the desired metadata devices", localDevice.Name)
+						metadataDevice = localDevice
+						break
+					} else if strings.HasPrefix(md, "/dev/") {
```

Comment:
Because of this `break` statement. The `else if` below is unnecessary. It would be cleaner as a new `if` statement.